mantis-server/mantis-server-worker-client/src/main/java/io/mantisrx/server/worker/client/MantisHttpClientImpl.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.mantisrx.server.worker.client;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInitializer;
+import java.util.ArrayList;
+import java.util.List;
+import mantis.io.reactivex.netty.channel.ObservableConnection;
+import mantis.io.reactivex.netty.client.ClientChannelFactory;
+import mantis.io.reactivex.netty.client.ClientConnectionFactory;
+import mantis.io.reactivex.netty.client.ClientMetricsEvent;
+import mantis.io.reactivex.netty.client.ConnectionPool;
+import mantis.io.reactivex.netty.client.ConnectionPoolBuilder;
+import mantis.io.reactivex.netty.metrics.MetricEventsSubject;
+import mantis.io.reactivex.netty.pipeline.PipelineConfigurator;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClientImpl;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClientRequest;
+import mantis.io.reactivex.netty.protocol.http.client.HttpClientResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Observable;
+
+public class MantisHttpClientImpl<I, O> extends HttpClientImpl<I, O> {
```

Comment:
Did you consider composition instead of inheritance? Can you potentially construct the HttpClientImpl outside and pass it to this constructor? This way a lot of the complex constructors will potentially go away.

Reply:
This class will still implement the HttpClient interface but delegate to it for most calls (you can use Lombok's @Delegate pattern for this) and decorate the methods that deal with the connection objects.

Reply:
I feel the current approach is simpler.

Reply:
Resolving without action.